### PR TITLE
isothermal and adiabatic wall BCs for reacting flows

### PR DESCRIPTION
**Pull request by DimAdam-01**
## Description

Immersed boundaries now support chemistry, with **isothermal and adiabatic wall** boundary conditions at immersed surfaces (`patch_ib(i)%isothermal`, `patch_ib(i)%twall`).

## Testing

2D thermal flat plate, bottom wall isothermal at 600 K:

image
T_vs_y

Regression test `CF9AFA00` (`2D -> Chemistry -> IBM Isothermal Wall`) passes on CPU **and** NVIDIA GPU (V100, OpenACC) to `1e-9`.

## Maintainer takeover

Finished by the maintainers on top of the original feature:

- **Correctness:** `present()`-guarded the new optional `t_step`/`stage` args in the cons→prim conversion; guarded the `Ys_IP`/`T_IP` writes on the bubbles/QBMM paths and prohibited the unsupported `chemistry + ib + (bubbles_euler | qbmm)` combo; device-initialize `ghost_points_index`/`pressure_ghost_point` (uninitialized on nvfortran, `ACC_SETUP_SFs` is Cray-only), allocate them only under `ib .and. chemistry`, and cover the moving-IB pressure store.
- **CPU/GPU divergence:** the IB spanned the full domain width, so corner ghost points read solid/buffer cells — ill-conditioned and CPU/GPU-divergent (~9%). Widening the IB past the domain fixes it (now ~`1e-11`); golden regenerated.
- Docs (`case.md` + `check_chemistry` PHYSICS_DOCS) added; Copilot review resolved.

## Checklist

- [x] Tests added/updated
- [x] Documentation updated

<strong>GPU changes</strong>

- [x] GPU results match CPU results
- [x] Tested on NVIDIA GPU

### CODE REVIEW

File: src/simulation/m_ibm.fpp [hunk 1]
```diff
@@ -248,13 +274,33 @@ contains
...
                 if (surface_tension) then
                     q_prim_vf(eqn_idx%c)%sf(j, k, l) = c_IP
                 end if
 
                 ! set the pressure
                 if (patch_ib(patch_id)%moving_ibm <= 1) then
                     q_prim_vf(eqn_idx%E)%sf(j, k, l) = pres_IP
+                    pressure_ghost_point%sf(j, k, l) = pres_IP
                 else
                     q_prim_vf(eqn_idx%E)%sf(j, k, l) = 0._wp
```
[the remark this answers is not in the document]
Comment (sbryngelson) [reply src/simulation/m_ibm.fpp:305]: Fixed in 5b99819b — `pressure_ghost_point` is now stored after both branches, so `moving_ibm>1` no longer leaves it stale.

File: src/common/m_variables_conversion.fpp [hunk 2]
```diff
@@ -361,6 +369,28 @@ contains
 
             $:GPU_UPDATE(device='[Res_vc, Re_idx, Re_size]')
         end if
+
+        if (ib) then
+            if (p > 0) then
+                @:ALLOCATE(ghost_points_index%sf(-buff_size:m+buff_size, -buff_size:n+buff_size, -buff_size:p+buff_size))
+                @:ALLOCATE(pressure_ghost_point%sf(-buff_size:m+buff_size, -buff_size:n+buff_size, -buff_size:p+buff_size))
+            else if (n > 0) then
```
[the remark this answers is not in the document]
Comment (sbryngelson) [reply src/common/m_variables_conversion.fpp]: Fixed in 5b99819b — both fields are now allocated only under `ib .and. chemistry`, with chemistry-guarded writes.

**Review by sbryngelson — APPROVED** [review]

**Review by sbryngelson — DISMISSED** [review]

**Review by sbryngelson — APPROVED** [review]
Approving. Isothermal/adiabatic IB walls for reacting flows; CF9AFA00 passes CPU + GPU (V100) to 1e-9.

Takeover: fixed the optional-arg/device-init correctness bugs, prohibited the unsupported chemistry+IB+bubbles/QBMM combo, root-caused and fixed the CPU/GPU divergence (IB was flush with the domain boundary → widened it), resolved the Copilot review, and added docs. precheck clean.

**Review by sbryngelson — APPROVED** [review]
Approving. Verified: CF9AFA00 passes CPU + GPU (V100) to 1e-9; precheck clean; review resolved.

**Comment by sbryngelson** [comment]
There's a bug in OMP for the chemistry + ibm case

**Comment by sbryngelson** [comment]
Closing, will reopen as features become required.
